woocommerce/woocommerce-ios
# [Bundle products in order] Bundle configuration form design updates

Part of #11259
Fixes #11235
Part of #10428


## Description


This PR includes a few visual updates in the bundle configuration form:

- #11259 - Use a large button at the bottom of the screen to save (used to be `Done` in the navigation bar)
  - Please note that there are other subtasks for the validation UX which I encountered some challenges and will update separately
- #11235 - Choose variation CTA is now only shown when the variable item is selected & quantity > 0
- #10428 - Don’t show variation button until product has been selected. Tidy up the layout of this button, and change it to “Choose variation”
  - In order to align the leading edge of the variation UI to the leading edge of the product row while there's a checkmark image beside it, I tried moving the product row into a VStack with the variation UI first. But then it's hard to vertically align the center of the checkmark image and the product row. I'm not aware of any way to align two views at different edges in different stacks in SwiftUI, and came up with the SwiftUI layout in the PR. Please feel free to suggest any easier ways than creating a spacer view to account for the checkmark image indentation
- Background color is now the same as the rows
- The check marks are no longer blocked by safe areas in landscape mode

## Testing instructions


Prerequisite: the store has [Product Bundles extension](https://woocommerce.com/products/product-bundles/) enabled, and also a bundle product with **at least 1 optional variable item** with non-empty variations and at least 1 non-variable item.

- Go to the Orders tab
- Tap `+` to create an order
- Tap `Add Products` and select the bundle product in the prerequisite --> the save CTA is now moved to the bottom that says `Save Configuration`. the optional variable item shouldn't have a CTA to choose a variation
- Tap to select the variable item --> a CTA `Choose variation` should be shown
- Tap `Choose variation` to choose a variation
- In the configuration form, the attributes should be shown that look like the expected design
- Configure the bundle so that it's valid
- Tap `Save Configuration`
- Tap `1 Product Selected` --> the bundle product should be added to the order with the expected configuration

---
- [x] @jaclync tests the quantity 0 case
- [x] @jaclync tests the landscape mode

## Screenshots


choose variation | update variation | quantity 0
-- | -- | --
<img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/490dec74-b73b-49c5-b537-49722da3c3e2" width="300" /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/f3672387-b3bd-42f6-a3c5-0f1a13e7dc6b" width="300" /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/1945542/63026061-e592-485b-b6c0-8b86f87d2f8d" width="300" />

---
